### PR TITLE
fix: list view (#565)

### LINKED ISSUE #565: List view is not working properly
Drive list view is not displaying files and folders properly.

### PR DESCRIPTION
**Pull request by tomicvladan**
Drive list view fix #565

### CODE REVIEW

File: src/components/Views/DriveListView/DriveListView.tsx [hunk 1]
```diff
@@ -24,18 +24,39 @@ const DriveListView: FC<DriveListViewProps> = ({
 }) => {
   const [page, setPage] = useState(0);
   const [rowsPerPage, setRowsPerPage] = useState(10);
-  const startIndex = page * rowsPerPage;
-  const endIndex = startIndex + rowsPerPage;
+
+  const folderLength = directories.length;
+  const folderStartIndex =
```
Comment (IgorShadurin) [inline src/components/Views/DriveListView/DriveListView.tsx:29]: const folderStartIndex = Math.min(page * rowsPerPage, folderLength);

File: src/components/Views/DriveListView/DriveListView.tsx [hunk 2]
```diff
@@ -24,18 +24,39 @@ const DriveListView: FC<DriveListViewProps> = ({
...
+    page * rowsPerPage > folderLength ? folderLength : page * rowsPerPage;
+  const folderEndIndex =
```
Comment (IgorShadurin) [inline src/components/Views/DriveListView/DriveListView.tsx:31]: const folderEndIndex = Math.min(folderStartIndex + rowsPerPage, folderLength);

**Review by IgorShadurin — CHANGES_REQUESTED** [review]

**Review by tamas6 — CHANGES_REQUESTED** [review]
tested, for me the list view/grid view worked as expected but the "sort" button did nothing when in list view, only when  in grid . 🧐

**Comment by tomicvladan** [comment]
I made a fix for sorting, so now it's working, but the component should be fixed later not to mutate state directly.

**Review by tfius — APPROVED** [review]

**Pull request merged**